packages/zowe-explorer/src/utils/ProfilesUtils.ts

```diff
@@ -329,21 +329,33 @@ export async function initializeZoweFolder(): Promise<void> {
 }
 
 export function writeOverridesFile() {
-    let content;
+    let fd: number;
     let fileContent: string;
     const settingsFile = path.join(getZoweDir(), "settings", "imperative.json");
-    if (fs.existsSync(settingsFile)) {
-        content = JSON.parse(fs.readFileSync(settingsFile).toString());
-        if (content && content?.overrides) {
-            if (content?.overrides?.CredentialManager === globals.PROFILE_SECURITY) {
+    try {
+        fd = fs.openSync(settingsFile, "r+");
+        fileContent = fs.readFileSync(fd, "utf-8");
+    } catch {
+        // If reading the file failed because it does not exist, then create it
+        // This should never fail, unless file system is read-only or the file
+        // was created by another process after first openSync call
+        fd = fs.openSync(settingsFile, "wx");
```

Comment:
line 342: Seems like if `openSync` fails here, it won't be caught. Do we have to wrap this in another `try/catch`?

It seems highly unlikely that this situation would occur, but we might still want to catch an exception here if one pops up.

Reply:
Good catch, I'm not sure in the `writeOverridesFile` method whether we want to catch all possible errors, or catch only the ones we need to handle and leave it to the caller to handle other errors.

If catching all possible errors is the desired behavior, then we should also catch and handle errors thrown by the `JSON.parse` call below.